spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraPartitionKeyRDD.scala

```diff
@@ -0,0 +1,125 @@
+package com.datastax.spark.connector.rdd
+
+import java.net.InetAddress
+
+
+import com.datastax.driver.core.{PreparedStatement, Session}
+import org.apache.spark.{Partitioner, TaskContext, Partition}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext._
+import scala.reflect.ClassTag
+
+import com.datastax.spark.connector.cql._
+import com.datastax.spark.connector._
+import com.datastax.spark.connector.writer._
+import com.datastax.spark.connector.rdd.reader._
+import com.datastax.spark.connector.rdd.partitioner.{EndpointPartition, ReplicaPartitioner}
+
+// O[ld] Is the type of the RDD we are Mapping From, N[ew] the type were are mapping too Old
+class CassandraPartitionKeyRDD[O, N] private[connector] (prev: RDD[O],
+                                      keyspaceName: String,
+                                      tableName: String,
+                                      connector: CassandraConnector,
+                                      columns: ColumnSelector = AllColumns,
+                                      where: CqlWhereClause = CqlWhereClause.empty,
+                                      readConf: ReadConf = ReadConf())
+                                      (implicit oldTag: ClassTag[O], newTag: ClassTag[N],
+                                       @transient rwf: RowWriterFactory[O], @transient rrf: RowReaderFactory[N])
+  extends CassandraRDD[N](prev.sparkContext, connector, keyspaceName, tableName, columns, where, readConf, prev.dependencies) {
+
+  //Make sure copy operations make new CPKRDDs and not CRDDs
+  override def copy(columnNames: ColumnSelector = columnNames,
+                   where: CqlWhereClause = where,
+                   readConf: ReadConf = readConf, connector: CassandraConnector = connector): CassandraPartitionKeyRDD[O,N] =
+    new CassandraPartitionKeyRDD[O,N](prev, keyspaceName, tableName, connector, columnNames, where, readConf)
+
+  private val converter = ReplicaMapper[O](connector, keyspaceName, tableName)
+
+  //We need to make sure we get selectedColumnNames before serialization so that our RowReader is
+  //built
+  private val singleKeyCqlQuery: (String) = {
+    logDebug("Generating Single Key Query Prepared Statement String")
+    val columns = selectedColumnNames.map(_.cql).mkString(", ")
+    val partitionWhere = tableDef.partitionKey.map(_.columnName).map(name => s"$name = :$name")
+    val filter = (where.predicates ++ partitionWhere).mkString(" AND ")
```

Comment:
I also realize this doesn't allow you to push in clustering column values, not sure if that's worth adding ... looking into it